src/connection/mod.rs

```diff
@@ -61,17 +50,22 @@ pub trait Connection {
     }
 }
 
-/// Extends [`Connection`] with a blocking `read` method.
+/// Extends [`Connection`] with `read` and `peek` methods.
 ///
-/// This trait exists as a convenient way to hook into `gdbstub`'s various
-/// byte-oriented APIs. It is _entirely optional_, and is not used in any
-/// `gdbstub` APIs. The `read` method this trait provides can be reimplemented
-/// using direct calls on a concrete type (e.g: using `std::io::Read`).
+/// This trait is used as part of `gdbstub`'s quickstart
+/// [`GdbStub::run`](crate::GdbStub::run) API.
 ///
 /// When the `std` feature is enabled, this trait is automatically implemented
 /// for [`TcpStream`](std::net::TcpStream) and
 /// [`UnixStream`](std::os::unix::net::UnixStream) (on unix systems).
+///
+/// [`gdbstub_run::Callbacks::read_byte`]:
+/// crate::gdbstub_run::Callbacks::read_byte
 pub trait ConnectionExt: Connection {
     /// Read a single byte.
     fn read(&mut self) -> Result<u8, Self::Error>;
+
+    /// Peek a single byte. This MUST be a **non-blocking** operation, returning
```

Comment:
FWIW it might be good to clarify in the docs here how peek should behave together with read. I assume if I do `peak -> read` or `peak -> peak -> read` it will return the "same byte" 2x in the first example and 3x in the second one? 
I 
was mainly wondering about this when `read()` was part of ConnectionExt but `peak()` was part of Connection what their relationship would be so maybe this is clear now.

Reply:
Yep, your intuition was right, though I agree that it never hurts to add more clarification to the docs.